apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -89,6 +90,8 @@ type ModelManagerV2 interface {
 	ModelStatus(req params.Entities) (params.ModelStatusResults, error)
 }
 
+type newCaasBrokerFunc func(args environs.OpenParams) (caas.Broker, error)
```

Comment:
Oh that's beautiful!! this is what I need to create in credential checks :) Thank you for making me see this!!!